src/amo/pages/UserProfileEdit/index.js

```diff
@@ -198,8 +200,25 @@ export class UserProfileEditBase extends React.Component<Props, State> {
       });
     }
 
-    if (wasUpdating && !isUpdating && !errorHandler.hasError()) {
+    const redirectToUserProfile = () => {
       history.push(`/${lang}/${clientApp}/user/${newUserId}/`);
+    };
+
+    if (wasUpdating && !isUpdating && !errorHandler.hasError()) {
+      let toPath = location.query.to;
+      if (toPath) {
```

Comment:
This might be paranoid but what about adding a check to make sure `typeof toPath` is `string`? If someone could figure out some prototype pollution (which is common) they might be able to pass an object to `history.push()` which could do all kinds of things.